### PR TITLE
Support additional APIs

### PR DESCRIPTION
**Pull request by gffking**
Also support updating patient PCP NPI

### CODE REVIEW

File: letter.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+type LetterSendToContactSpeciality struct {
+	ID   int64  `json:"id"`
+	Name string `json:"name"`
+}
+
+type FindLettersOptions struct {
+	*Pagination
+
+	DocumentDateGT  time.Time `url:"document_date_gt,omitempty"`
```
Comment (elliehastings) [inline letter.go:78]: For all the gt/lt values - is it `document_date_gt` or `document_date__gt`? The docs seem to have the latter

File: non_visit_note.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	ID          int64     `json:"id"`
+	Author      int64     `json:"author"`       //: 64,
+	Text        string    `json:"text"`         //: "note is here",
+	Version     int64     `json:"version"`      //: 2,
+	UpdatedDate time.Time `json:"updated_date"` //: "2022-05-15T13:50:09Z"
+}
+
+type NonVisitNoteDocument struct {
+	ID       int64                        `json:"id"`
+	Document NonVisitNoteDocumentDocument `json:"document"`
```
Comment (elliehastings) [inline non_visit_note.go:53]: 😅

**Review by elliehastings — COMMENTED** [review]
lgtm, just the one question about the underscore formats for Letters!

**Review by fterrag — APPROVED** [review]

**Comment by fterrag** [comment]
@gffking LGTM pending Ellie's feedback

File: contact.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	"strconv"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type ContactServicer interface {
+	Find(ctx context.Context, opts *FindContactsOptions) (*Response[[]*Contact], *http.Response, error)
```
[the remark this answers is not in the document]
Comment (gffking) [reply contact.go:16]: Oops, I copied/pasted without looking at the docs. I see there's both a List and a Search API for contacts. I'll need to look a little deeper, but I think I might need the List API. Will update the name of this method accordingly.

File: letter.go (hunk 1 above)
Comment (gffking) [reply letter.go:78]: Missed the extra underscore. Will fix.

**Pull request merged**